### PR TITLE
steamtinkerlaunch: init at 11.11

### LINKED ISSUE #186826: Steam Tinker Launch
**Project description**

Steam Tinker Launch is a Linux wrapper tool for use with the Steam client which allows customizing and start tools and options for games quickly on the fly (see [Features](https://github.com/frostworx/steamtinkerlaunch#Features))

**Metadata**

* homepage URL: https://github.com/frostworx/steamtinkerlaunch
* source URL: https://github.com/frostworx/steamtinkerlaunch
* license: GPL-3.0 
* platforms: linux

### PR DESCRIPTION
**Pull request by urandom2**
###### Description of changes
Fixes #186826

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/games/steamtinkerlaunch/default.nix [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      vkBasalt
+      wget
+      wine
+      winetricks
+      xdg-utils
+      xdotool
+      xorg.xprop
+      xorg.xrandr
+      xorg.xwininfo
+      yad
```
Comment (SuperSandro2000) [inline pkgs/tools/games/steamtinkerlaunch/default.nix:94]: Do we really need **all** of this including rsync, strace and gdb?

File: pkgs/tools/games/steamtinkerlaunch/default.nix [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  src = fetchFromGitHub {
+    owner = "sonic2kk";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-rWENtgV6spokzkhnmrrzsAQ19dROJ50ofEulU5Jx5IE=";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  PREFIX = "\${out}";
```
Comment (SuperSandro2000) [inline pkgs/tools/games/steamtinkerlaunch/default.nix:52]: can we use installFlags instead?

**Comment by urandom2** [comment]
this also suffers from the `.xxx-wrapped` as arg0 bug, let me know what the canonical resolution to that is:
```
Tue 08 Nov 2022 12:42:17 UTC INFO - main - No arguments provided. See '.steamtinkerlaunch-wrapped --help' for possible command line parameters
```

File: pkgs/tools/games/steamtinkerlaunch/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/tools/games/steamtinkerlaunch/default.nix:93]: Yes, we should reduce that list to a sensible default. I would suggest to remove all the debugging utilties.

**Comment by SuperSandro2000** [comment]
> this also suffers from the `.xxx-wrapped` as arg0 bug, let me know what the canonical resolution to that is:
> 
> ```
> Tue 08 Nov 2022 12:42:17 UTC INFO - main - No arguments provided. See '.steamtinkerlaunch-wrapped --help' for possible command line parameters
> ```

The error message nearly suggests that you did not provide any arguments but you can change argv0 with the argv0 argument. See https://github.com/NixOS/nixpkgs/blob/d291728eee55d5befa60421e84fb3e78e0ccc284/pkgs/build-support/setup-hooks/make-binary-wrapper/make-binary-wrapper.sh for details

**Comment by urandom2** [comment]
I tried that, it appears to be the bash script introspecting its filename.

File: pkgs/tools/games/steamtinkerlaunch/default.nix (hunk 1 above)
Comment (urandom2) [reply pkgs/tools/games/steamtinkerlaunch/default.nix:93]: what is the canonical way to pin optional dependencies, a second package called `steamtinkerlaunch-full`?

**Comment by SuperSandro2000** [comment]
> I tried that, it appears to be the bash script introspecting its filename.

Then we should rather patch that code.

**Comment by urandom2** [comment]
ok, so this just looks like a bash behaviour, see [upstream docs](https://www.gnu.org/software/bash/manual/html_node/Special-Parameters.html#index-0), you cannot lie about argv0; as such, I will patch the script to hardcode the binary name

File: pkgs/tools/games/steamtinkerlaunch/default.nix [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  version = "11.11";
+
+  src = fetchFromGitHub {
+    owner = "sonic2kk";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-rWENtgV6spokzkhnmrrzsAQ19dROJ50ofEulU5Jx5IE=";
+  };
+
+  # hardcode PROGCMD because #150841
```
Comment (SuperSandro2000) [inline pkgs/tools/games/steamtinkerlaunch/default.nix:29]: Links should ideally be absolute

**Pull request merged**